### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by hwindsor**
Permissions framework for Colin to do his security work off.

### CODE REVIEW

**Comment by bluemellophone** [comment]
@hwindsor Can you please rebase this PR onto the latest version of develop?  There are merge conflicts that could not be automatically resolved.

File: tests/modules/submissions/resources/test_ensure_submission.py [hunk 1]
```diff
@@ -53,3 +53,60 @@ def test_ensure_clone_submission_by_uuid(
...
+            auth_scopes=(
+                'submissions:read',
+                'assets:read',
+            ),
+        ):
+            submission_response = flask_app_client.get(
+                '/api/v1/submissions/%s' % test_clone_submission_uuid
+            )
+            asset_response = flask_app_client.get(
+                '/api/v1/assets/3abc03a8-39c8-42c4-bedb-e08ccc485396'
```
Comment (bluemellophone) [inline tests/modules/submissions/resources/test_ensure_submission.py:80]: We may want to parameter the UUID in the future

**Review by bluemellophone — APPROVED** [review]
LGTM

**Review by colinwkingen — APPROVED** [review]
Looks good to me.

File: tests/modules/submissions/resources/test_ensure_submission.py (hunk 1 above)
Comment (hwindsor) [reply tests/modules/submissions/resources/test_ensure_submission.py:80]: Agreed. I thought that as I copy & bodged it from the test above :)
Replace test_clone_submission_uuid with test_clone_submission that returns the UUID to be cloned and the UUIDs of the assets that it should contain, I guess. I'll have a look at that.

**Pull request merged**